app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,28 @@
+class CommentsController < ApplicationController
+  before_action :require_login
+
+  def create
+    @comment = Comment.new(comment_params)
+
+    if @comment.save
+      flash[:notice] = "Your comment was saved!"
+      redirect_to trail_path(@comment.trail_id)
+    else
+      flash[:notice] = "Comment was not saved!"
+      redirect_to trail_path(@comment.trail_id)
+    end
+  end
+
+  protected
+  def require_login
+    unless current_user
+      flash[:notice] = "You must be logged in to comment!"
+      redirect_to root_path
+    end
+  end
```

Comment:
I can imagine that the `require_login` method could be useful in many of your controllers. If you move this code to the `ApplicationController` all inheriting controllers will be able to use this method.